Modules/_sre.c

```diff
@@ -2787,37 +2785,27 @@ PyMODINIT_FUNC PyInit__sre(void)
     m = PyModule_Create(&sremodule);
     if (m == NULL)
         return NULL;
-    d = PyModule_GetDict(m);
 
-    x = PyLong_FromLong(SRE_MAGIC);
-    if (x) {
-        PyDict_SetItemString(d, "MAGIC", x);
-        Py_DECREF(x);
+    if (PyModule_AddIntConstant(m, "MAGIC", SRE_MAGIC) < 0) {
+        return NULL;
     }
 
-    x = PyLong_FromLong(sizeof(SRE_CODE));
-    if (x) {
-        PyDict_SetItemString(d, "CODESIZE", x);
-        Py_DECREF(x);
+    if (PyModule_AddIntConstant(m, "CODESIZE", sizeof(SRE_CODE)) < 0) {
+        return NULL;
     }
 
-    x = PyLong_FromUnsignedLong(SRE_MAXREPEAT);
-    if (x) {
-        PyDict_SetItemString(d, "MAXREPEAT", x);
-        Py_DECREF(x);
+    if (PyModule_AddIntConstant(m, "MAXREPEAT", SRE_MAXREPEAT) < 0) {
```

Comment:
It is not correct. `PyModule_AddIntConstant()` takes a `long` argument, but `SRE_MAXREPEAT` can be `2**32-1` or `2**63-1`.

Reply:
Right, they're unsigned. I missed that.

Reply:
Would it be acceptable to add `PyModule_AddUnsignedIntConstant` to `Python/modsupport.c`?

Reply:
Fixed in ab13ae6

Reply:
> Would it be acceptable to add `PyModule_AddUnsignedIntConstant` (and possibly also `PyModule_AddUnsignedIntMacro`)

This is a different issue (maybe there is already an open issue?). We can try and see how much code will benefit from this.

Reply:
I'll check bpo and do some grepping around the code. If there's more than a handful of users, it may be worth it. I any case, I did not mean to include such a modification in this PR.

Reply:
A quick search shows 9 users in the standard library:
Modules/_stat.c: 3
Modules/_sre.c: 2
Modules/socketmodule.c: 1
Modules/_ssl.c: 1
Modules/_testcapimodule.c: 2